tools/gmhazard_scripts/db_creation/empirical_db/calc_emp_flt.py

```diff
@@ -46,54 +40,132 @@ def calculate_flt(
+        cur_site_df = site_df.iloc[(ix * 1000) : (ix + 1) * 1000]
+
+        start_time = time.time()
+        if n_procs == 1:
+            im_data = []
+            for _, cur_site in cur_site_df.iterrows():
+                im_data.append(
+                    _process_site(
+                        cur_site,
+                        site_source_db_ffp,
+                        fault_df,
+                        rupture_df,
+                        keep_sigma,
+                        ims,
+                        psa_periods,
+                        imdb_dict,
+                        nhm_data,
+                        tect_type_model_dict,
+                    )
                 )
-                common.calculate_emp_site(
-                    ims,
-                    psa_periods,
-                    imdb_dict,
-                    fault_df,
-                    rupture_df,
-                    distance_store,
-                    nhm_data,
-                    site.vs30,
-                    site.z1p0 if hasattr(site, "z1p0") else None,
-                    site.z2p5 if hasattr(site, "z2p5") else None,
-                    site.name,
-                    tect_type_model_dict_ffp,
-                    max_dist,
-                    keep_sigma_components=keep_sigma,
-                    use_directivity=distance_store.has_station_directivity_data(site.name)
+        else:
+            with mp.Pool(n_procs) as p:
+                im_data = p.starmap(
+                    _process_site,
+                    [
+                        (
+                            cur_site,
+                            site_source_db_ffp,
+                            fault_df,
+                            rupture_df,
+                            keep_sigma,
+                            ims,
+                            psa_periods,
+                            imdb_dict,
+                            nhm_data,
+                            tect_type_model_dict,
+                        )
+                        for _, cur_site in cur_site_df.iterrows()
+                    ],
                 )
-            else:
-                print(
-                    f"Skipping site {(site_df.index.get_loc(site_index) + 1)} / {n_stations} - Rank: {rank}"
-                )
-    if is_master:
-        common.write_data_and_close(
+
+        print(f"Computed data for sites {ix * 1000} - {(ix + 1) * 1000}, "
+              f"took {time.time() - start_time:.2f} seconds; writing to DB")
+        # for cur_site_name, cur_im_data in im_data:
+        #     common.write_result_to_db(cur_im_data, imdb_dict, cur_site_name)
+        write_result_to_db(im_data, imdb_dict)
+        print("wtf")
+
+    common.write_data_and_close(
+        imdb_dict,
+        nhm_ffp,
+        rupture_df,
+        site_df,
+        vs30_ffp,
+        psa_periods,
+        ims,
+        tect_type_model_dict_ffp,
+        rupture_lookup=rupture_lookup,
+    )
+
```

Comment:
May want to run black on this file as some of these gaps don't seem right

Reply:
Just did, don't think there was anything. These are all functions, so there are always 2 empty lines between them.

Reply:
There was 1 empty line gap for this function, so figured black wasn't run for this file.